.github/workflows/e2e.yml

```diff
@@ -0,0 +1,127 @@
+
+env:
+  FORCE_COLOR: true
+  CARGO_HOME: "${{ github.workspace }}/cargo"
+
+jobs:
+  snapshot:
+    uses: ./.github/workflows/snapshot.yml
+
+  fluence-js-tests:
+    needs:
+      - snapshot
+
+    uses: fluencelabs/fluence-js/.github/workflows/tests.yml@master
+    with:
+      avm-version: ${{ needs.snapshot.outputs.avm-version }}
+
+  fluence-js-snapshot:
+    needs:
+      - snapshot
+
+    uses: fluencelabs/fluence-js/.github/workflows/snapshot.yml@master
+    with:
+      avm-version: ${{ needs.snapshot.outputs.avm-version }}
+
+  aqua-snapshot:
+    needs:
+      - fluence-js-snapshot
+
+    uses: fluencelabs/aqua/.github/workflows/snapshot.yml@main
+    with:
+      fluence-js-version: ${{ needs.fluence-js-snapshot.outputs.fluence-js-version }}
+
+  rust-peer:
+    name: "Build rust-peer snapshot"
+    runs-on: builder
+
+    outputs:
+      digest: "${{ steps.build.outputs.digest }}"
+
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - name: Checkout rust-peer
+        uses: actions/checkout@v3
+        with:
+          repository: fluencelabs/rust-peer
+
+      - name: Setup rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: Use local dependencies
+        run: |
+          sed -i 's#\(avm-server =\) .*#\1 { git = "https://github.com/fluencelabs/aquavm", branch = "${{ github.head_ref }}" }#g' Cargo.toml
+
+      - name: Run cargo build
+        run: cargo build --release -p particle-node
+
+      - name: Install cargo-nextest
+        uses: baptiste0928/cargo-install@v1
+        with:
+          crate: cargo-nextest
+          version: "0.9"
+
+      - name: Run cargo nextest
+        run: cargo nextest run --release --all-features --no-fail-fast --retries 10 --test-threads 10
+
+      - name: Import secrets
+        uses: hashicorp/vault-action@v2.4.1
+        with:
+          url: https://vault.fluence.dev
+          path: jwt/github
+          role: ci
+          method: jwt
+          jwtGithubAudience: "https://github.com/fluencelabs"
+          jwtTtl: 300
+          secrets: |
+            kv/docker-registry/basicauth/ci username | DOCKER_USERNAME ;
+            kv/docker-registry/basicauth/ci password | DOCKER_PASSWORD
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          registry: docker.fluence.dev
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Build and push snapshot
+        id: build
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: true
+          file: .github/snapshot/Dockerfile
+          tags: |
+            docker.fluence.dev/aquavm:${{ github.head_ref }}
+          labels: |
+            sha=${{ github.event.pull_request.head.sha }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
Why not move it to `snapshot.yml` in rust peer repo and call it like
```
  rust-peert:
     uses: fluencelabs/rust-peer/.github/workflows/snapshot.yml@main
     with:
       rust-peer-foo-bar: ...
```

Reply:
Yes, that is a good idea, maybe in another PR?